File: api/_lib/ocr.py
"""
Claude Vision 기반 OCR 서비스.
비용 절감을 위해 기본 비활성화 (ENABLE_OCR=false).
프론트엔드에서는 🔒 '준비중' 표시로 노출.
"""

import os

ENABLE_OCR = os.environ.get("ENABLE_OCR", "false").lower() == "true"


def is_enabled() -> bool:
    return ENABLE_OCR


def recognize_product(image_bytes: bytes) -> dict:
    """상품 사진 → 브랜드/제품명/스펙 인식."""
    if not ENABLE_OCR:
        return {
            "enabled": False,
            "message": "🔒 상품 인식 기능은 준비중입니다.",
        }

    # import base64
    # import anthropic
    # client = anthropic.Anthropic(api_key=os.environ["ANTHROPIC_API_KEY"])
    # b64 = base64.standard_b64encode(image_bytes).decode("utf-8")
    # msg = client.messages.create(
    #     model="claude-sonnet-4-6",
    #     max_tokens=512,
    #     messages=[{
    #         "role": "user",
    #         "content": [
    #             {"type": "image", "source": {
    #                 "type": "base64", "media_type": "image/jpeg", "data": b64}},
    #             {"type": "text",
    #              "text": "이 상품 사진에서 브랜드, 제품명, 스펙(용량/수량)을 JSON으로 반환"},
    #         ],
    #     }],
    # )
    # return {"enabled": True, "result": msg.content[0].text}
    return {"enabled": False, "message": "Not implemented"}


def parse_receipt(image_bytes: bytes) -> dict:
    """영수증 사진 → 구매 품목/가격/쇼핑몰 추출."""
    if not ENABLE_OCR:
        return {
            "enabled": False,
            "message": "🔒 영수증 인식 기능은 준비중입니다.",
        }

    # import base64
    # import anthropic
    # client = anthropic.Anthropic(api_key=os.environ["ANTHROPIC_API_KEY"])
    # b64 = base64.standard_b64encode(image_bytes).decode("utf-8")
    # msg = client.messages.create(
    #     model="claude-sonnet-4-6",
    #     max_tokens=1024,
    #     messages=[{
    #         "role": "user",
    #         "content": [
    #             {"type": "image", "source": {
    #                 "type": "base64", "media_type": "image/jpeg", "data": b64}},
    #             {"type": "text",
    #              "text": "영수증에서 판매처, 각 품목의 이름/수량/단가/금액을 JSON으로 반환"},
    #         ],
    #     }],
    # )
    # return {"enabled": True, "items": msg.content[0].text}
    return {"enabled": False, "message": "Not implemented"}

File: api/_lib/ocr.js
// Claude Vision OCR — 🔒 준비중 스텁. 비용 절감을 위해 기본 비활성화.
// ENABLE_OCR=true 로 설정하면 아래 실제 호출부를 구현하면 됨.

function enabled() {
  return (process.env.ENABLE_OCR || 'false').toLowerCase() === 'true';
}

export function parseOrderTextEnabled() {
  return enabled();
}

// Claude API 연동부 — ENABLE_OCR=true 일 때만 실제 호출.
// 현재는 stub 으로 null 을 리턴 → logic.parseOrderText 가 simple parser fallback.
export async function parseOrderTextWithClaude(/* text */) {
  if (!enabled()) {
    return { parser: 'simple', items: null };
  }
  // TODO: Claude API 호출
  // import Anthropic from '@anthropic-ai/sdk';
  // const client = new Anthropic({ apiKey: process.env.ANTHROPIC_API_KEY });
  // const msg = await client.messages.create({
  //   model: 'claude-sonnet-4-6',
  //   max_tokens: 1024,
  //   messages: [{
  //     role: 'user',
  //     content: '다음 쿠팡 주문내역에서 각 라인의 브랜드/제품명/규격을 ' +
  //              'JSON 배열 [{name, brand, spec}, ...] 로만 추출:\n' + text,
  //   }],
  // });
  // const parsed = JSON.parse(msg.content[0].text);
  // return { parser: 'claude', items: parsed };
  return { parser: 'simple', items: null };
}

export function recognizeProduct(/* imageBytes */) {
  if (!enabled()) {
    return { enabled: false, message: '🔒 상품 인식 기능은 준비중입니다.' };
  }
  // TODO: Claude API 호출
  // import Anthropic from '@anthropic-ai/sdk';
  // const client = new Anthropic();
  // const msg = await client.messages.create({ model: 'claude-sonnet-4-6', ... });
  return { enabled: false, message: 'Not implemented' };
}

export function parseReceipt(/* imageBytes */) {
  if (!enabled()) {
    return { enabled: false, message: '🔒 영수증 인식 기능은 준비중입니다.' };
  }
  return { enabled: false, message: 'Not implemented' };
}
